Sources/StreamChat/Workers/ChannelUpdater.swift

```diff
@@ -882,9 +882,10 @@ extension ChannelUpdater {
         }
     }
 
-    @discardableResult func update(channelQuery: ChannelQuery, isInRecoveryMode: Bool) async throws -> ChannelPayload {
-        try await withCheckedThrowingContinuation { continuation in
-            update(channelQuery: channelQuery, isInRecoveryMode: isInRecoveryMode, onChannelCreated: nil) { result in
+    @discardableResult func update(channelQuery: ChannelQuery) async throws -> ChannelPayload {
+        let useCreateEndpoint: ((ChannelId) -> Void)? = channelQuery.cid == nil ? { _ in } : nil
+        return try await withCheckedThrowingContinuation { continuation in
+            update(channelQuery: channelQuery, isInRecoveryMode: false, onChannelCreated: useCreateEndpoint) { result in
```

Comment:
I do not need `onChannelCreated` closure which just returns `cid` but I have to set it with a dummy closure since depending on if this, we choose the endpoint in the `ChannelUpdater`.